packages/engine-formula/src/functions/logical/switch/index.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2023-present DreamNum Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ErrorType } from '../../../basics/error-type';
+import type { BaseValueObject } from '../../../engine/value-object/base-value-object';
+import type { ArrayValueObject } from '../../../engine/value-object/array-value-object';
+import { ErrorValueObject } from '../../../engine/value-object/base-value-object';
+import { BaseFunction } from '../../base-function';
+
+export class Switch extends BaseFunction {
+    override minParams = 3;
+
+    override maxParams = 255;
+
+    override calculate(expression: BaseValueObject, ...casesAndResults: BaseValueObject[]): BaseValueObject {
```

Comment:
Need to support array formulas, like
![image](https://github.com/dream-num/univer/assets/26371161/37517d85-5721-4af0-a184-b0d1e6065f02)